ops/testing.py

```diff
@@ -93,6 +97,16 @@ def framework(self) -> framework.Framework:
         """Return the Framework that is being driven by this Harness."""
         return self._framework
 
+    @property
+    def _resource_dir(self) -> pathlib.Path:
+        if self._lazy_resource_dir is not None:
+            return self._lazy_resource_dir
+
+        self.__resource_dir = tempfile.TemporaryDirectory()
+        self._lazy_resource_dir = pathlib.Path(self.__resource_dir.name)
+        self._finalizer = weakref.finalize(self, self.__resource_dir.cleanup)
+        return self._lazy_resource_dir
```

Comment:
Having discussed this with others on the team, I think we'd rather change this to having a

```
class Harness:
   def cleanup(self):
   ...
```

So that you would do:

```
harness = Harness(...)
self.addCleanup(harness.cleanup)
```

In tests, rather than using finalizers. However, we can land your code and then do the change to add a cleanup method ourselves.

Reply:
Sounds good